### DISCUSSION TITLE
Logos and Images not displaying, only broken icons

### DISCUSSION THREAD

**Discussion by zjorz** [neos-sdi/adfsmfa#231] [General]
When triggering MFA and using this MFA adapter on ADFS 2019, all images from the MFA adapter are not displaying. Instead an icon is shown as if the image is not available. Right clicking that icon and opening it in new tab does show the image/code.
What an I missing?
regards,
Jorge

**Comment by zjorz** [comment]
For another product I had configured security headers (Content-Security-Policy) in ADFS.
Running the following solved the issue: Set-AdfsResponseHeaders -RemoveHeaders "Content-Security-Policy"

**Comment by SkorpionMars** [comment]
Might be useful to someone, the problem with images not loading was missing **img-src 'self' data:;** in Content-Security-Policy
Get-AdfsResponseHeaders | Select -ExpandProperty ResponseHeaders

After executing the command Set-AdfsResponseHeaders -SetHeaderName "Content-Security-Policy" -SetHeaderValue "default-src 'self' 'unsafe-inline' 'unsafe-eval'; img-src 'self' data:;" everything worked. For some reason, when installing the MFA, this parameter was not added.

Before adding, I advise you to look at what you have here Get-AdfsResponseHeaders | Select -ExpandProperty ResponseHeaders are already there and add what you need to the end.

**Reply by redhook62** [reply]
@SkorpionMars, @zjorz 

Thanks for the info, super cool.
However, these values must be present by default.
We don't want to change ADFS settings as a matter of principle.

regards

**Reply by SkorpionMars** [reply]
@redhook62 Hi, I understand you.

Perhaps the parameter was set by default and changes were made by the administrators themselves in the course of work. I didn’t have such a problem in the Sandbox and everything worked without problems, unlike the Production.

In any case, I think it will be useful for others if they suddenly encounter this.

regards

**Reply by redhook62** [reply]
Thanks  very much
